app/src/main/java/org/fossasia/openevent/app/core/auth/login/LoginFragment.java

```diff
@@ -64,6 +64,12 @@ public void onStart() {
         loginFragmentViewModel.getProgress().observe(this, this::showProgress);
         loginFragmentViewModel.getError().observe(this, this::showError);
         loginFragmentViewModel.getEmailList().observe(this, this::attachEmails);
+        loginFragmentViewModel.getIsTokenSent().observe(this, (value) -> {
+            getFragmentManager().beginTransaction()
+                .setCustomAnimations(R.anim.enter_from_left, R.anim.exit_from_right)
+                .replace(R.id.fragment_container, new ResetPasswordFragment())
+                .commit();
```

Comment:
What this would do is create a new `ResetPasswordFragment` everytime you rotate the screen and the `value` is set to true.

Reply:
Even if it doesn't then also it's not good for readability. A `SingleActionLiveData` has been added specifically for this purpose. Check how we are opening dashboard/event list in the `MainActivity` and the `EventViewModel` of the `main` module.

Reply:
> What this would do is create a new ResetPasswordFragment everytime you rotate the screen and the value is set to true.

Didn't understand this. Once the value is set, ResetPasswordFragment will be replaced.
> Even if it doesn't then also it's not good for readability

Not sure how using a custom class improves readability.
> A `SingleActionLiveData` has been added specifically for this purpose.

`SingleEventLiveData` I think is being used for events which can be triggered repeatedly, but should notify only once. But in this case, the 'I forgot my password' button can be clicked only once in the lifecycle. Once it's clicked, the fragment will be destroyed.
`SingleEventLiveData` seemed to be right in case of adding fragment instead of replacing, but even that is not true I think, because then the `SingleEventLiveData` won't send be able to send anymore notifications, which is crucial for continual functioning of the 'I forgot my password' button even after fragment recreation.

Reply:
That's not what SingleEventLiveData is used for @mishuvs 

It just doesn't retransmit the last value on rotation. That's it

Reply:
Okay @iamareebjamal , but what I am not able to understand is what does it has to do with rotation?
The value is false all the time -> Button clicked -> value sets to  true -> ResetPasswordFragment replaced and eventually destroyed -> if back to login, the value again starts with false
Should I anyway use SingleEventLiveData?